pkg/workflows/dontime/plugin.go

```diff
@@ -66,52 +66,38 @@ func (p *Plugin) Observation(_ context.Context, outctx ocr3types.OutcomeContext,
 		p.lggr.Errorf("failed to unmarshal previous outcome in Observation phase")
 	}
 
-	// Collect up to batchSize unexpired requests
 	requests := map[string]int64{} // Maps executionID --> seqNum
-	for batchOffset := 0; batchOffset < p.store.requests.Len() && len(requests) < p.batchSize; {
-		batch, err := p.store.requests.RangeN(batchOffset, p.batchSize)
-		if err != nil {
-			p.lggr.Errorf("failed to get request batch at offset %d: %v", batchOffset, err)
-			batchOffset += p.batchSize
+	timeoutCheck := time.Now()
+	for _, req := range p.store.GetRequests() {
```

Comment:
I'd like to highlight the risk again - if this ever gets larger than max Query size, then the plugin could get stuck forever (until node restart). Request expiry should prevent that from happening.